### ISSUE TITLE
Add a feature to close the running instance of grunt via RESTful API call

### ISSUE THREAD

**Issue by wangyx2005** [Mayo-QIN/grunt#10]
Can you add a REST endpoint for stopping the running instance of grunt. I know I can stop it via `docker` command, but it would be great if the instance can stop via API call. Or is it already supported I just miss it?

**Comment by potis** [comment]
Hi @wangyx2005 , 
Can you provide some more information on what exactly are you planning to implement or an example use case? Would you like to stop a process running within the docker or gracefully shut down the docker and stop all the queued processes.

**Comment by wangyx2005** [comment]
Hi, sorry for reply so late. I missed your reply before. I am working with Dr. Sharma to build a platform to enable image processing pipeline. A sample user case is like this: a image processing algorithm container is built on grunt, another 'master' container issue images to the grunt container to process images via RESTful API. Once 'master' container is done with the grunt container, it can issue another API call to stop the grunt container.

Another not related question, i do not know if i should open a new issue, I saw a new feature 'zip download working directory' is added. Could you add some documentation on how to use that? Is it just use the same as before? If the *commandLine* argument generates a folder, grunt will autocratically zip it? I am sorry I should look into the code to find myself. I just not very familiar with GO yet.

Thank you.

**Comment by potis** [comment]
Regarding the fist request (stopping the running instance): This is hard to implement since there is no login mechanism/  control on who is accessing the grunt. I would propose to implement the shut down mechanism as an end point in your .yml file.  For example in python this rest point could execute:

```
import os
os.system("shutdown now -h")
```

Regarding your second request: You have to just visit the appropriate end point after your job has finished. 

`
/rest/job/{id}/zip
`
Hope this helps.

**Comment by wangyx2005** [comment]
@potis  Thank you for your reply. I think implementing the shut down mechanism as an end point is a great alternative. Thank you, I will try that.
